src/librustdoc/html/static/main.js

```diff
@@ -467,22 +467,7 @@ function hideThemeButtonState() {
     function expandSection(id) {
         var elem = document.getElementById(id);
         if (elem && isHidden(elem)) {
-            var h3 = elem.parentNode.previousElementSibling;
-            if (h3 && h3.tagName !== "H3") {
-                h3 = h3.previousElementSibling; // skip div.docblock
-            }
-
-            if (h3) {
-                var collapses = h3.getElementsByClassName("collapse-toggle");
-                if (collapses.length > 0) {
-                    // The element is not visible, we need to make it appear!
-                    collapseDocs(collapses[0], "show");
-                }
-                // Open all ancestor <details> to make this element visible.
-                openParentDetails(h3.parentNode);
-            } else {
-                openParentDetails(elem.parentNode);
-            }
+            openParentDetails(elem.parentNode);
```

Comment:
All of expandSection can become just this:

```
openParentDetails(document.getElementById(id);
```

Because if the element absent, or not hidden, that's a no-op.